lib/bolt/cli.rb

```diff
@@ -390,6 +390,14 @@ def handle_parser_errors
     def execute(options)
       if options[:mode] == 'plan' || options[:mode] == 'task'
         begin
+          if Gem.win_platform?
+            # Windows 'fix' for openssl behaving strangely. Prevents very slow operation
+            # of random_bytes later when establishing winrm connections from a Windows host.
+            # See https://github.com/rails/rails/issues/25805 for background.
+            require 'openssl'
```

Comment:
Given this will 'pause' bolt for a few seconds on older rubies, perhaps emit a debug message saying "Warming up OpenSSL" or something to that effect

Reply:
Why do you say it’ll take a few seconds on older Ruby? My finding was that the time to do this was negligible, and made later calls much faster.

Reply:
Ruby 2.3.1

Without this PR
![bolt-without](https://user-images.githubusercontent.com/5819622/35131195-a84568b4-fcff-11e7-92b8-ab282a36c06f.gif)

Note the runtime of ~10s

With this PR (But I've enabled it for `bolt command` just for this test)
![bolt-with](https://user-images.githubusercontent.com/5819622/35131221-c052c064-fcff-11e7-9f01-2282146729fc.gif)

Note the runtime is now ~5s BUT it takes about 4-5 seconds for any output to be shown.  This is in debug mode too.

Reply:
I'm not able to reproduce that... I'm using Windows 10 Enterprise (x64), Ruby 2.3.3 from RubyInstaller. Where are you getting Ruby 2.3.1 from?

I'm running in Powershell and using Measure-Command. I get no increase runtime from this change. The command also runs in 5 seconds rather than 10 for me.

Reply:
* Ruby 2.3.1 is from https://rubyinstaller.org/downloads/archives/

* Using PowerShell measure-command will of course show no change.  In my examples above the total runtime is always ~10s.  This PR shifts the delay earlier but the delay has to be incurred somewhere

* The VM I'm connecting to is a default-ish Windows installation with default WinRM settings.  Pooler has allowunecrypted and basic auth enabled.  When using bolt (with this PR) across the VPN into pooler, I was getting runtimes of 4.0 seconds.  Compared to pooler-to-pooler connection of 3.5-4.0 seconds. So that's reasonable.

* There are differences in my laptop and pooler

Laptop:
Intel64 Family 6 Model 60 Stepping 3 GenuineIntel ~2201 Mhz (Intel Speed Step will change the speed as well varied between 2.0 and 2.9GHZ)
8GB RAM
Windows 10 1709 (Fall Creators Update)

Pooler:
Intel64 Family 6 Model 62 Stepping 4 GenuineIntel ~2800 Mhz
4GB RAM
Windows 10 1703 (Creators Update)

Given that ruby is "mostly" single threaded, raw CPU speed will make a difference

---
But this doesn't change the UX of bolt seeming to do nothing for 3-5 seconds on startup.

Reply:
I'm not able to reproduce this particular scenario. I've tried with Ruby 2.3.1 on a win-10-ent-x86_64 vmpooler image as well as a Windows 10 1709 image in VMWare Fusion on my laptop. In both cases I don't see any particularly slow behavior with `bolt command` with vanilla Bolt 0.14.0 and after adding this PR.

The only thing remotely similar I saw was that, after a fresh start of Windows 10, the first invocation of Bolt was slightly slower to start logging. Subsequent runs were normal.